app/renderer/js/utils/domain-util.js

```diff
@@ -101,54 +102,80 @@ class DomainUtil {
-				\nIf you have a valid certificate please add it from Settings>Organizations and try to add the organization again.
-				\nUnless you have a good reason to believe otherwise, you should not proceed.
-				\nYou can click here if you'd like to proceed with the connection.`;
-
-				dialog.showMessageBox({
-					type: 'warning',
-					buttons: ['Yes', 'No'],
-					defaultId: 1,
-					message: certErrorMessage,
-					detail: certErrorDetail
-				}, response => {
-					if (response === 0) {
-						// set ignoreCerts parameter to true in case user responds with yes
-						serverConf.ignoreCerts = true;
-						this.getServerSettings(domain, serverConf.ignoreCerts).then(serverSettings => {
-							resolve(serverSettings);
-						}, error => {
-							if (error === `${domain} does not have any organizations added.\
-							\nPlease contact your server administrator.`) {
-								reject(error);
-							}
-							resolve(serverConf);
-						});
-					} else {
-						reject('Untrusted certificate');
+	async checkCertError(domain, serverConf, error, silent) {
+		if (silent) {
+			// since getting server settings has already failed
+			return serverConf;
+		} else {
+			// Report error to sentry to get idea of possible certificate errors
+			// users get when adding the servers
+			logger.reportSentry(new Error(error));
+			const certErrorMessage = `Do you trust certificate from ${domain}? \n ${error}`;
+			const certErrorDetail = `The organization you're connecting to is either someone impersonating the Zulip server you entered, or the server you're trying to connect to is configured in an insecure way.
+			\nIf you have a valid certificate please add it from Settings>Organizations and try to add the organization again.
+			\nUnless you have a good reason to believe otherwise, you should not proceed.
+			\nYou can click here if you'd like to proceed with the connection.`;
+
+			const response = await dialog.showMessageBox({
+				type: 'warning',
+				buttons: ['Yes', 'No'],
+				defaultId: 1,
+				message: certErrorMessage,
+				detail: certErrorDetail
+			});
+			if (response === 0) {
+				// set ignoreCerts parameter to true in case user responds with yes
+				serverConf.ignoreCerts = true;
+				try {
+					return await this.getServerSettings(domain, serverConf.ignoreCerts);
+				} catch (err) {
+					if (error === `${domain} does not have any organizations added.\
+					\nPlease contact your server administrator.`) {
+						throw new Error(error);
 					}
-				});
+					return serverConf;
+				}
+			} else {
+				throw new Error('Untrusted certificate.');
 			}
-		});
+		}
+	}
+
+	async checkOggFile({ serverConf, domain, whitelistDomains, ignoreCerts, silent }) {
+		const oggCheck = {
+			url: domain + '/static/audio/zulip.ogg',
+			...RequestUtil.requestOptions(domain, ignoreCerts)
+		};
+		try {
+			const response = await requestPromise(oggCheck);
+			if (response.statusCode === 200) {
+				// getting serverSettings has already failed
```

Comment:
Why does this comment states something failed inside a 200 request condition?
Also I recommend checking range of good HTTP status i.e. 200 to but not including 400 since cached responses return 302 Not Modified.

Reply:
> Why does this comment states something failed inside a 200 request condition?

I wanted to clarify why we're returning `serverConf` instead of `serverSettings`. Since we already tried retrieving the latter and got an error, we need to return `serverConf`. 

> I recommend checking range of good HTTP status i.e. 200 to but not including 400 since cached responses return 302 Not Modified.

That's a good catch, thanks. I'll update the PR>

Reply:
Can change the comment to explain this better and include the statement you mentioned.

Reply:
@abhigyank marking this as resolved because we've decided to do away with the .ogg file check entirely, so there's no need to change the comments.